src/core/evaluator.js

```diff
@@ -2007,9 +2007,8 @@ class PartialEvaluator {
                   localColorSpaceCache,
                 })
                 .then(function (colorSpace) {
-                  if (colorSpace) {
-                    stateManager.state.fillColorSpace = colorSpace;
-                  }
+                  stateManager.state.fillColorSpace =
+                    colorSpace || ColorSpace.singletons.gray;
```

Comment:
I believe that we could fix all regressions with the following changes.
```suggestion
                  if (colorSpace) {
                    stateManager.state.fillColorSpace = colorSpace;
                  } else if (!stateManager.state.patternFillColorSpace) {
                    stateManager.state.fillColorSpace = ColorSpace.singletons.gray;
                  }
```